ui/src/alert-configuration/ConfigureAlerting.test.tsx

```diff
@@ -0,0 +1,1275 @@
+    patchAlertmanagerConfig.mockClear();
+    // Setting defaultHidden will disable visibility check on aria-label
+    // and speed up the test that use `getByRole`.
+    configure({ defaultHidden: true });
+  });
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  const selectors = {
+    enableConfiguration: () =>
+      screen.getByLabelText(/Enable Email Notification Configuration \*/i),
+    host: () => screen.getByLabelText(/SMTP Host \*/i),
+    port: () => screen.getByLabelText(/SMTP Port \*/i),
+    enableTls: () => screen.getByLabelText(/Enable SMTP Over TLS/i),
+    authSelect: {
+      label: () => screen.getByLabelText(/SMTP Auth \*/i).parentElement,
+      authClick: () => screen.getByLabelText(/SMTP Auth \*/i),
+      optionNoAuth: () =>
+        screen.getByRole('option', { name: /NO AUTHENTICATION/i }),
+      optionLogin: () => screen.getByRole('option', { name: /LOGIN/i }),
+      optionCramMd5: () => screen.getByRole('option', { name: /CRAM-MD5/i }),
+      optionPlain: () => screen.getByRole('option', { name: /PLAIN/i }),
+    },
+    username: () => screen.getByLabelText(/Username \*/i),
+    password: () => screen.getByLabelText(/Password \*/i),
+    identity: () => screen.getByLabelText(/identity \*/i),
+    secret: () => screen.getByLabelText(/secret \*/i),
+    sender: () => screen.getByLabelText(/Sender Email Address \*/i),
+    recipient: () => screen.getByLabelText(/Recipient Email Addresses \*/i),
+    receiveResolved: () =>
+      screen.getByLabelText(/Enable Receive Resolved Alerts/i),
+    sendTestingEmailButton: () =>
+      screen.getByRole('button', { name: /send a test email/i }),
+    cancelButton: () => screen.getByRole('button', { name: /cancel/i }),
+    saveButton: () => screen.getByRole('button', { name: /save|saving.../i }),
+  };
+
+  it('render default value when the form is not defined', async () => {
+    render(<ConfigureAlerting />);
+    await waitForLoadingToFinish();
+
+    await act(async () => {
+      await userEvent.click(selectors.authSelect.authClick());
+    });
+
+    expect(selectors.enableConfiguration()).not.toBeChecked();
+    expect(selectors.host()).toHaveValue('');
+    expect(selectors.port()).toHaveValue(25);
+    expect(selectors.enableTls()).not.toBeChecked();
+
+    expect(selectors.authSelect.label()).toHaveTextContent('NO AUTHENTICATION');
+    expect(selectors.authSelect.optionNoAuth()).toBeInTheDocument();
+    expect(selectors.authSelect.optionLogin()).toBeInTheDocument();
+    expect(selectors.authSelect.optionCramMd5()).toBeInTheDocument();
+    expect(selectors.authSelect.optionPlain()).toBeInTheDocument();
+
+    expect(selectors.sender()).toHaveValue('');
+    expect(selectors.recipient()).toHaveValue('');
+    expect(selectors.receiveResolved()).not.toBeChecked();
+
+    expect(selectors.sendTestingEmailButton()).toBeEnabled();
+    expect(selectors.cancelButton()).toBeEnabled();
+    expect(selectors.saveButton()).toBeEnabled();
```

Comment:
I think the save button should be disabled when the form is not "dirty" 😬

Reply:
Thx, I just add this logic.